tokio-stream/tests/stream_partition.rs

```diff
@@ -0,0 +1,46 @@
+use tokio_stream::{self as stream, StreamExt};
+use tokio_test::{assert_pending, assert_ready_eq, task};
+
+mod support {
+    pub(crate) mod mpsc;
+}
+
+#[tokio::test]
+async fn partition() {
+    let stream = stream::iter(0..6);
```

Comment:
This test only tests a situation where the items are ready. A test for when the values are not ready should be added.